configure.ac

```
@@ -61,21 +63,12 @@ PKG_CHECK_MODULES([ZLIB],[zlib])
 AC_ARG_WITH([libmpg123],[AS_HELP_STRING([--without-libmpg123],
 	[Disable improved MP3 support provided by libmpg123. Uses SDL_mixer instead which results in noise or crashes for some MP3s. @<:@default=auto@:>@])])
 AS_IF([test "x$with_libmpg123" != "xno"],[
-	PKG_CHECK_MODULES([MPG123],[libmpg123],[AC_DEFINE(HAVE_MPG123,[1],[Disable improved MP3 support provided by libmpg123])],[auto_mpg123=0])
+	PKG_CHECK_MODULES([MPG123],[libmpg123],[AC_DEFINE(HAVE_MPG123,[1],[Enable improved MP3 support provided by libmpg123])],[auto_mpg123=0])
 ])
-# wildmidi has no pkg-config support (yet), this aims to make the same features available
-AC_ARG_VAR([WILDMIDI_CFLAGS], [C compiler flags for WILDMIDI])
-AC_ARG_VAR([WILDMIDI_LIBS], [linker flags for WILDMIDI])
-AC_ARG_WITH([libwildmidi],[AS_HELP_STRING([--with-libwildmidi],
-	[Enable Midi support provided by libwildmidi. @<:@default=no@:>@])])
-AS_IF([test "x$with_libwildmidi" = "xyes"],[
-	AS_IF([test -z "$WILDMIDI_CFLAGS" -a -z "$WILDMIDI_LIBS"],[
-		AC_CHECK_LIB([WildMidi],[WildMidi_Init],[AC_DEFINE(HAVE_WILDMIDI,[1],[Enable Midi support provided by libwildmidi]) WILDMIDI_LIBS=[-lWildMidi]],[auto_wildmidi=0])
-	],[
-		AC_MSG_CHECKING([for WILDMIDI])
-		AC_MSG_RESULT([assuming yes])
-		AC_DEFINE(HAVE_WILDMIDI,[1],[Enable Midi support provided by libwildmidi])
-	])
+AC_ARG_WITH([libwildmidi],[AS_HELP_STRING([--without-libwildmidi],
+	[Disable Midi support provided by libwildmidi. @<:@default=auto@:>@])])
+AS_IF([test "x$with_libwildmidi" != "xno"],[
+	PKG_CHECK_MODULES([WILDMIDI],[wildmidi],[AC_DEFINE(HAVE_WILDMIDI,[1],[Enable Midi support provided by libwildmidi])],[auto_wildmidi=0])
```

Setting minimum version might be useful for completeness (and maybe for other modules, if apply):

```suggestion
	PKG_CHECK_MODULES([WILDMIDI],[wildmidi >= 0.4.1],[AC_DEFINE(HAVE_WILDMIDI,[1],[Enable Midi support provided by libwildmidi])],[auto_wildmidi=0])
```

For testing this, it should generate a proper `Requires:` in the .pc file.

---

I have added pkg-support in WildMidi for 0.4.1, so this check is useless.